build.sh

```diff
@@ -640,23 +639,13 @@ function processSource {
   BUILD_TYPE="$2"
   cp -p  entities/out/entities.inc "$HTML_CACHE"
   cp -p  entities/out/entities-dtd.url "$HTML_CACHE"
-  if [[ $PROCESS_WITH_RUST == "true" ]]; then
-    if hash html-build 2>/dev/null; then
-      html-build <"$HTML_SOURCE/$SOURCE_LOCATION" >"$HTML_TEMP/source-whatwg-complete"
-    else
-      CARGO_ARGS=()
-      $VERBOSE && CARGO_ARGS+=( --verbose )
-      $QUIET && CARGO_ARGS+=( --quiet )
-      cargo run "${CARGO_ARGS[@]}" --release <"$HTML_SOURCE/$SOURCE_LOCATION" >"$HTML_TEMP/source-whatwg-complete"
-    fi
+  if hash html-build 2>/dev/null; then
+    html-build <"$HTML_SOURCE/$SOURCE_LOCATION" >"$HTML_TEMP/source-whatwg-complete"
   else
-    if $VERBOSE; then
-      perl .pre-process-main.pl --verbose < "$HTML_SOURCE/$SOURCE_LOCATION" > "$HTML_TEMP/source-expanded-1"
-    else
-      perl .pre-process-main.pl < "$HTML_SOURCE/$SOURCE_LOCATION" > "$HTML_TEMP/source-expanded-1"
-    fi
-    perl .pre-process-annotate-attributes.pl < "$HTML_TEMP/source-expanded-1" > "$HTML_TEMP/source-expanded-2" # this one could be merged
-    perl .pre-process-tag-omission.pl < "$HTML_TEMP/source-expanded-2" | perl .pre-process-index-generator.pl > "$HTML_TEMP/source-whatwg-complete" # this one could be merged
+    CARGO_ARGS=()
+    $VERBOSE && CARGO_ARGS+=( --verbose )
+    $QUIET && CARGO_ARGS+=( --quiet )
+    cargo run "${CARGO_ARGS[@]}" --release <"$HTML_SOURCE/$SOURCE_LOCATION" >"$HTML_TEMP/source-whatwg-complete"
```

Comment:
```suggestion
    cargo run "${CARGO_ARGS[@]-''}" --release <"$HTML_SOURCE/$SOURCE_LOCATION" >"$HTML_TEMP/source-whatwg-complete"
```
This changes eliminates the error in my environment

Reply:
Give it a try now, I fixed in a slightly different way.

Reply:
I'll force-push this fix onto the current main so that the diff here can be just un-indenting.